openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -483,8 +487,8 @@ def update_library(
         if library_type != COMPLEX:
             for block in get_library_blocks(library_key):
                 if block.usage_key.block_type != library_type:
-                    raise ValidationError(
-                        {'type': _('You can only set a library to that type if all existing blocks are of that type.')},
+                    raise IncompatibleTypesError(
+                        _('You can only set a library to that type if all existing blocks are of that type.'),
```

Comment:
Please include the desired library type, offending block ID, and offending block type. That info could help us use the logs  to track down a bug related to this, if one ever came up.